lib/rock/rest_api.rb

```diff
@@ -0,0 +1,142 @@
+require 'eventmachine'
+require 'rack'
+require 'thin'
+require 'orocos'
+require 'rack/static'
+require 'sprockets'
+require 'rock/webapp/tasks'
+
+module Rock
+    
+class RestApi
+        
+    def self.exec(argv = [])
+        
+        thin_host = '0.0.0.0'
+        thin_port = 9292
+         
+        name_server_hosts=[]
+        cors_hosts=[]
+            
+        syskit = false
+
+        options = OptionParser.new do |opt|
+            opt.on('--host host', String, 'the host of the name server that should be contacted (default to localhost)') do |host|
+                name_server_hosts << host
+            end
+            opt.on('--bind host', String, 'the host the server should bind to (default to 0.0.0.0)') do |host|
+                thin_host = host
+            end
+            opt.on('--port port', Integer, "the server's port (default to #{thin_port})") do |port|
+                thin_port = port
+            end
+            opt.on('--enable-cors hosts', Array, "enables Cross-Origin Resource Sharing for sites loaded from the parameter may include localhost host:port, file://. Also multiple values seperated by ','") do |host|
+                cors_hosts = host
+            end
+            opt.on('--enable-syskit',"load syskit support") do
+                syskit=true
+            end
+            opt.on('--config-dir dir', String, 'directory where the configuration can be found') do |dir|
+                config_dir = dir
+            end
+        end
+        options.parse argv
+        
+        
+        if syskit
+            #includes roby app that changes a log path, must be included before Orocos.initialize
+            require 'rock/webapp/syskit'
+        end
+        
+        #if no --host option is given: add the default host 'localhost'
+        if name_server_hosts.empty?
+            name_server_hosts << 'localhost'
+        end
+        
+        #use the first entry as default name server and remove it from list
+        Orocos::CORBA.name_service.ip = name_server_hosts.first
+        name_server_hosts.shift
+        
+        #add the remaining additional name servers
+        name_server_hosts.each do |host_address|
+            Orocos.name_service << Orocos::CORBA::NameService.new(host_address)
+        end
+        
+        #when started from a rock ruby script, Orocos might already be initialized
+        if !Orocos.initialized?
```

Comment:
Don't. Expect the caller to have initialize Orocos, and check it's the case at the beginning of the method.